### DISCUSSION TITLE
I am new to rust how can I run this app?

### DISCUSSION THREAD

**Discussion by aliscie** [AppFlowy-IO/AppFlowy#425] [Technical help wanted] [answered]
Should I run `npm run` or `cargo start` to launch the app and use it?

**Comment by AkechiShiro** [comment, marked as answer]
Hi @aliscie, please make sure to read the README first before opening an issue (unless you read the README and there is a mistake somewhere in it), there were links to the documentation explaining how to install, how to build, how to contribute : https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/environment-setup/building-on-linux

As I'm not a contributor of this project I can't close the issue but if you have no other question, then someone from AppFlowy could close the issue I believe.

**Comment by aliscie** [comment]
I tried that didn't work for me also I am using pycharm.

**Comment by AkechiShiro** [comment]
Alright, could you please explain in details the error (logs, what command did you run if you ran any?) in order for someone to help you ?
Or you only opened the source code in your Charm IDE ? (And what Charm IDE are you using from Jetbrains I assume)

**Comment by aliscie** [comment]
@AkechiShiro when I ran `cargo make flowy_dev` I got this error 
```py
[cargo-make][1] INFO - Running Task: install_diesel
    Updating crates.io index
     Ignored package `diesel_cli v1.4.1` is already installed, use --force to override
[cargo-make][1] INFO - Running Task: install_protobuf_compiler
Install protoc_plugin (Dart)
/var/folders/v3/4llt971130g6vvbq53m6k84w0000gn/T/apple/fsio/4mYy3lHHKn.sh: line 4: dart: command not found
[cargo-make][1] ERROR - Error while executing command, exit code: 127
[cargo-make][1] WARN - Build Failed.
[cargo-make] INFO - Running Task: legacy-migration
[cargo-make] INFO - Running Task: restore-crate-type
[cargo-make] ERROR - Task error detected, exit code: 1
[cargo-make] WARN - Build Failed.
(base) apples-MBP:frontend apple$ 
```

**Comment by AkechiShiro** [comment]
@aliscie you probably have an issue with flutter, could you run `flutter doctor` and show its output ?
Because the `dart` command is coming from flutter.
What distro linux are you on ? How did you install flutter and did you make any PATH modification after installing flutter ?
Finally can you run `dart --version` in your shell ?

**Comment by annieappflowy** [comment]
@aliscie  Ali, are you able to run the app?

**Reply by AlenSci** [reply]
Hi, Yes I was able to, also  Now I have my own notion clone https://github.com/aliscie/autodox-rust/blob/main/README.md
